### PR TITLE
INS-3408: proper flow pulse usage, better handling

### PR DESCRIPTION
**Pull request by sexyhamster**
**- What I did**
- moved to pulse from flow
- removed currentexecutor check

**- How I did it**

**- How to verify it**
run tests

### CODE REVIEW

File: logicrunner/logicrunner.go [hunk 1]
```diff
@@ -311,42 +311,15 @@ func contextWithServiceData(ctx context.Context, data *payload.ServiceData) cont
 
 func (lr *LogicRunner) AddUnwantedResponse(ctx context.Context, msg insolar.Payload) error {
 	m := msg.(*payload.ReturnResults)
-	currentPulse, err := lr.PulseAccessor.Latest(ctx)
+	done, err := lr.WriteController.Begin(ctx, flow.Pulse(ctx))
 	if err != nil {
-		return errors.Wrap(err, "failed to get current pulse")
+		if err == writecontroller.ErrWriteClosed {
+			inslogger.FromContext(ctx).Debug("")
```
Comment (ruz) [inline logicrunner/logicrunner.go]: :)

Comment (sexyhamster) [reply logicrunner/logicrunner.go]: fixed

File: logicrunner/logicrunner.go [hunk 2]
```diff
@@ -311,42 +311,14 @@ func contextWithServiceData(ctx context.Context, data *payload.ServiceData) cont
...
 func (lr *LogicRunner) AddUnwantedResponse(ctx context.Context, msg insolar.Payload) error {
 	m := msg.(*payload.ReturnResults)
-	currentPulse, err := lr.PulseAccessor.Latest(ctx)
+	done, err := lr.WriteController.Begin(ctx, flow.Pulse(ctx))
 	if err != nil {
-		return errors.Wrap(err, "failed to get current pulse")
+		if err == writecontroller.ErrWriteClosed {
+			return flow.ErrCancelled
+		}
+		return errors.Wrap(err, "could acquire writecontroller lock")
```
Comment (bigbes) [inline logicrunner/logicrunner.go]: couldn't acquire

Comment (ruz) [reply logicrunner/logicrunner.go]: %)

**Pull request merged**